h5py/_hl/files.py

```diff
@@ -344,6 +358,9 @@ def __init__(self, name, mode=None, driver=None,
             that can fit in rdcc_nbytes bytes. For maximum performance, this
             value should be set approximately 100 times that number of
             chunks. The default value is 521.
+        track_order
+            Track dataset/group/attribute creation order under root group
+            if True. If None use global default h5.get_config().track_order.
```

Comment:
Should we clarify that this only has an effect if the file is created?

@tacaswell do you also want to throw an error if this is passed in combination with mode `'r'` or `'r+'`? I think I recall from another PR that you're keen not to allow options when they don't do anything.